src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -1810,14 +1830,17 @@ Status ColorControlServer::moveToSaturationCommand(EndpointId endpoint,
  * @param commandData Struct containing the parameters of the command.
  * @return Status::Success when successful,
  *         Status::InvalidCommand when a step size of 0 or an unknown SaturationStepMode is provided
- *         Status::UnsupportedEndpoint when the provided endpoint doesn't correspond with a saturation transition state.
+ *         Status::UnsupportedEndpoint when the provided endpoint doesn't correspond with a saturation transition state,
+ *         Status::ConstraintError when a command parameters is outside its defined value range.
  */
 Status ColorControlServer::stepSaturationCommand(EndpointId endpoint, const Commands::StepSaturation::DecodableType & commandData)
 {
     MATTER_TRACE_SCOPE("stepSaturation", "ColorControl");
     // Confirm validity of the step mode and step size received
     VerifyOrReturnValue(commandData.stepMode != SaturationStepMode::kUnknownEnumValue, Status::InvalidCommand);
     VerifyOrReturnValue(commandData.stepSize != 0, Status::InvalidCommand);
+    // Command Parameters constraint checks:
+    VerifyOrReturnValue(commandData.transitionTime <= UINT8_MAX, Status::ConstraintError);
```

Comment:
I am surprised you are not getting "this comparison is always true" errors, since the type of that field is `uint8_t`....

But more generally: if the constraint is `all`, why is there a check at all?